test/src/discover_tasks_test.dart

```diff
@@ -147,6 +148,40 @@ main() {
     });
   });
 
+  group('execute returns value', () {
+    grinderTest('task from method', () {}, (MockGrinderContext ctx) {
+      var annotated =
+          discoveryGood.discoverDeclaration(goodLib.declarations[#method], {});
+      var task = annotated.task;
+      expect(task.name, 'method');
```

Comment:
[This test](https://github.com/google/grinder.dart/blob/master/test/grinder_test.dart#L197) already checks that `grinder.start` waits for a future.